File: langgraph/data/semantic_equiv/88.py
def arrange_sequence(sequence):
    if not sequence:
        return []
    
    reverse_flag = (sequence[0] + sequence[-1]) % 2 == 0
    return sorted(sequence, reverse=reverse_flag)

# method name: sort_array -> arrange_sequence
# parameter name: array -> sequence
# changed the condition to `if not sequence:` for checking empty list
# extracted the reverse condition into a variable `reverse_flag` for clarity
# used the `sorted` function with the `reverse` parameter directly

File: langgraph/result/qwen/3b-eq2js/88.js
function arrangeSequence(sequence) {
    if (!sequence) {
        return [];
    }

    const reverseOrder = (sequence[0] + sequence[sequence.length - 1]) % 2 === 0;
    return sequence.slice().sort((a, b) => reverseOrder ? b - a : a - b);
}